### PR TITLE
Refactor/launch args sim nav

### PR DESCRIPTION
**Pull request by AdarshKaran**

### CODE REVIEW

**Comment by AdarshKaran** [comment]
I have made the necessary changes with my latest commit "Updated_Refactored launch arguments"
[https://youtu.be/GAQIj6Hk1Ys](url)
this is the youtube link for the video- this is just a temporary solution. If the video is finalized, I can directly upload it to the repository

**Comment by AdarshKaran** [comment]
https://github.com/neobotix/neo_simulation2/assets/112091438/edc4225d-3c8a-4718-850c-b0e49e6d1125

File: launch/navigation.launch.py [hunk 1]
```diff
@@ -3,65 +3,153 @@
...
     param_file_name = 'navigation.yaml'
     param_dir = LaunchConfiguration(
         'params_file',
         default=os.path.join(
             get_package_share_directory('neo_simulation2'),
-            'configs/'+MY_NEO_ROBOT,
+            'configs/'+my_neo_robot,
             param_file_name))
-
+
```
Comment (padhupradheep) [inline launch/navigation.launch.py]: remove additional indents

File: launch/simulation.launch.py [hunk 2]
```diff
@@ -56,5 +52,80 @@ def generate_launch_description():
...
         )
+    
+    spawn_entity = Node(
+        package='gazebo_ros', 
+        executable='spawn_entity.py',
+        arguments=['-entity', my_neo_robot,'-file', robot_description_urdf,], 
+        output='screen')
+    
+    # if use_robot_state_pub is true, start the robot state publisher node
+    if use_robot_state_pub:
```
Comment (padhupradheep) [inline launch/simulation.launch.py]: why is there an `if` condition? robot_state_publisher should be always there..

File: launch/simulation.launch.py [hunk 3]
```diff
@@ -56,5 +52,80 @@ def generate_launch_description():
...
+    
+    context_arguments = [my_neo_robot_arg, my_neo_env_arg, use_sim_time_arg, use_robot_state_pub_arg]
+    opq_function = OpaqueFunction(function=launch_setup, 
+                                  args=context_arguments)
+
+    ld.add_action(opq_function)
 
-    return LaunchDescription([spawn_entity, start_robot_state_publisher_cmd, teleop, gazebo])
\ No newline at end of file
+    return ld
```
Comment (padhupradheep) [inline launch/simulation.launch.py]: add an empty extra line..

**Review by padhupradheep — CHANGES_REQUESTED** [review]

File: launch/simulation.launch.py (hunk 2 above)
Comment (AdarshKaran) [reply launch/simulation.launch.py]: i will remove it anna...it was there in the older version of the code so I added a it as a launch argument...but now I'll completely remove it

File: launch/navigation.launch.py (hunk 1 above)
Comment (AdarshKaran) [reply launch/navigation.launch.py]: the indentation seems to be fine in my code editor anna...i removed 2 empty lines but apart from that where do u feel it is wrong?

**Comment by Keshava2k01** [comment]
I tried recreating it and the simulation worked perfectly in gazebo but getting some errors in the navigation.launch file, its not working correctly, it stops at "creating bond timer..." and when Rviz is launched separately the map is not loading properly as well

**Comment by padhupradheep** [comment]
pointing Rolling, tested on 22.04! 

Good work @AdarshKaran  ..

Thanks @Keshava2k01 for testing it!

**Pull request merged**